staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go

```diff
@@ -136,6 +152,9 @@ const (
 	NamesAccepted CustomResourceDefinitionConditionType = "NamesAccepted"
 	// Terminating means that the CustomResourceDefinition has been deleted and is cleaning up.
 	Terminating CustomResourceDefinitionConditionType = "Terminating"
+	// Valid means the CRD does pass validation. K8S objects are not validated on the GET path and it is
+	// possible to have invalid objects on downgrade/upgrade paths.
+	Valid CustomResourceDefinitionConditionType = "Valid"
```

Comment:
Isn't validation an invariant of every object in etcd we always try to guarantee? How can it become invalid?

Reply:
With @liggitt's comment https://github.com/kubernetes/kubernetes/issues/65790#issuecomment-404509214 in mind, I don't think this is a path worth pursuing. We have the very same problem of data-loss for any rollback which takes away features and fields that we used before rollback. Our cure: if you want to rollback, wait one Kube version iteration more until your rollback target version supports what you use.

Also note that all future conversion features will start as alpha with a feature gate. I.e. we have to plan in advance the transition from alpha to beta, and possibly a rollback to alpha. In that case, we can keep serving and understanding new validation features in a graceful way, even when rolling back to kube where the feature is still alpha.

Reply:
I should remove this, was remnant of another solution.

> Also note that all future conversion features will start as alpha with a feature gate.

I do not disagree but still thinking about this :)

Generally I think it is still good to have this in 1.11.2. I commented on the related issue.